### PR TITLE
Added label smoothing to cross entropy loss

### PR DESCRIPTION
**Pull request by tjaffri**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add label smoothing to cross entropy loss

## Modification

Adds a call to the newly added pytorch label_smoothing capability: https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html#torch.nn.functional.cross_entropy

## BC-breaking (Optional)

Requires pytorch 1.10.0 or higher. The label_smoothing parameter was added in this release: https://pytorch.org/docs/1.10.0/generated/torch.nn.functional.cross_entropy.html#torch.nn.functional.cross_entropy and did not exist before.

## Use cases (Optional)

Add label smoothing to cross entropy loss in your config.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by ZwwWayne** [comment]
Hi @tjaffri ,
Thanks for your kind PR. Could you sign the CLA so that we finally could merge this PR after review?
image

**Comment by ZwwWayne** [comment]
Hi @tjaffri ,
Since this PR require the new pytorch feature, could you add the version check? For example, only allow label_smoothing to be set when pytorch version is > 1.10.

**Comment by tjaffri** [comment]
@ZwwWayne feedback appreciated on how I did the python version condition check. If you have a different preferred pattern, I would appreciate a pointer. For now I'm just ignoring label_smoothing if the python version is less than 1.10.

**Comment by tjaffri** [comment]
@ZwwWayne gentle ping. Please let me know if I can improve this PR further.

File: mmdet/models/losses/cross_entropy_loss.py [hunk 1]
```diff
@@ -32,19 +33,32 @@ def cross_entropy(pred,
...
     # element-wise losses
-    loss = F.cross_entropy(
-        pred,
-        label,
-        weight=class_weight,
-        reduction='none',
-        ignore_index=ignore_index)
+
+    if torch.__version__ < '1.10':
+        loss = F.cross_entropy(
```
Comment (jbwang1997) [inline mmdet/models/losses/cross_entropy_loss.py:47]: This comparison may be wrong. we can use the `digit_version` in `mmcv.utils.version_utils`

File: mmdet/models/losses/cross_entropy_loss.py [hunk 2]
```diff
@@ -32,19 +33,32 @@ def cross_entropy(pred,
...
+            pred,
+            label,
+            weight=class_weight,
+            reduction='none',
+            ignore_index=ignore_index)
```
Comment (jbwang1997) [inline mmdet/models/losses/cross_entropy_loss.py:52]: When torch version is lower than 1.10, we can assert `label_smooth !=0` to avoid unexpected behavior of useless `label_smooth`.

**Comment by jbwang1997** [comment]
Hello @tjaffri. 

Thanks for your valuable contribution. I have left some comments above. Please remember to check.

Best wishes，

**Comment by OpenMMLab-Assistant003** [comment]
Hi @tjaffri！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @tjaffri
